mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -432,7 +438,14 @@ impl State for RefundTakerPayment {
 
         let refund_fut = watcher_ctx
             .taker_coin
-            .send_taker_payment_refund_preimage(&watcher_ctx.data.taker_payment_refund_preimage);
+            .send_taker_payment_refund_preimage(SendWatcherRefundsPaymentArgs {
+                payment_tx: &watcher_ctx.data.taker_payment_refund_preimage,
+                swap_contract_address: &None,
+                secret_hash: &watcher_ctx.data.secret_hash,
+                other_pubkey: &watcher_ctx.verified_pub,
+                time_lock: watcher_ctx.taker_locktime() as u32,
+                swap_unique_data: &[],
+            });
```

Comment:
Here too, please create a structure instance before the `send_taker_payment_refund_preimage` method call. It will reduce some code lines.

And [here in `impl State for ValidateTakerFee`](https://github.com/KomodoPlatform/atomicDEX-API/blob/ethereum-swap-watchers/mm2src/mm2_main/src/lp_swap/swap_watcher.rs#L173).  Please, create an `WatcherValidateTakerFeeInput` instance before `watcher_validate_taker_fee` method call.